app/src/main/java/com/example/evently/ui/auth/AuthActivity.java

```diff
@@ -97,19 +101,84 @@ private void showRegisterForm() {
-                            startActivity(transition);
-                            finish();
+                            String email = bundle.getString("email");
+                            String name = bundle.getString("name");
+                            String phone = bundle.getString("phone");
+
+                            // Stores a new account
+                            AccountDB accountDB = new AccountDB();
+                            accountDB.fetchAccount(
+                                    email,
+                                    optionalAccount -> {
+                                        if (optionalAccount.isEmpty()) {
+                                            hasRegisterForm = true;
+                                            // If user is not found, we need to create an account
+                                            // for them with the new information
+                                            Account newAccount = new Account(
+                                                    email, name, Optional.ofNullable(phone), email);
+                                            accountDB.storeAccount(newAccount);
+                                            startActivity(transition);
+                                            finish();
+                                        } else {
+                                            // NOTE: If the bundle contains an email that already
+                                            // exists in the DB, show an error and prompt the user
+                                            // to login instead of registering.
+                                            FirebaseAuth.getInstance().signOut();
+                                            Toast.makeText(
+                                                            this,
+                                                            "Account already registered: Please login",
+                                                            Toast.LENGTH_SHORT)
+                                                    .show();
+
+                                            binding.login.setVisibility(View.VISIBLE);
+                                            binding.registerForm.setVisibility(View.VISIBLE);
+                                            hasRegisterForm = false;
+                                            // The following code is from the stack overflow
+                                            // question below:
+                                            // Title: "Remove old Fragment from fragment manager"
+                                            // Author: "khouloud mejdoub"
+                                            // Response Author: "Yashdeep Patel"
+                                            // Link: https://stackoverflow.com/a/22474821
+                                            getSupportFragmentManager()
```

Comment:
Store the fragment above where it's initially spawned so you don't need to find it again.